vowpalwabbit/bfgs.cc

```diff
@@ -106,20 +106,16 @@ const char* curv_message = "Zero or negative curvature detected.\n"
 
 void zero_derivative(vw& all)
 { //set derivative to 0.
-  uint32_t length = 1 << all.num_bits;
-  size_t stride_shift = all.reg.stride_shift;
-  weight* weights = all.reg.weight_vector;
-  for(uint32_t i = 0; i < length; i++)
-    weights[(i << stride_shift) +W_GT] = 0;
+  weight_vector& weights = all.wv;
+  for (weight_vector::iterator i = weights.begin(W_GT); i != weights.end(W_GT); ++i)
+	  *i = 0;
```

Comment:
You know something is going right when the number of lines of code decreases :-)